src/main/scala/net/ripe/rpki/publicationserver/repository/DataFlusher.scala

```diff
@@ -265,16 +265,18 @@ class DataFlusher(conf: AppConfig)(implicit val system: ActorSystem)
     IOStream.string("</publish>\n", stream)
   }
 
-  def withAtomicStream(targetFile: Path)(f : HashingSizedStream => Unit) = {
-    val tmpFile = Files.createTempFile(targetFile.getParent, "", ".xml")
-    val tmpStream = new HashingSizedStream(new FileOutputStream(tmpFile.toFile))
-    try {
-      f(tmpStream)
-      Files.move(tmpFile, targetFile.toAbsolutePath, StandardCopyOption.REPLACE_EXISTING, StandardCopyOption.ATOMIC_MOVE)
-      tmpStream.summary
-    } finally {
-      tmpStream.close()
-      Files.deleteIfExists(tmpFile)
+  def withAtomicStream(target: (Path, FileAttributes))(f : HashingSizedStream => Unit) = target match {
+    case (targetFile, attrs) => {
```

Comment:
I think this is matter of preference, maybe it's simpler to not pass pair of path & file attributes, and add new params instead i.e: `withAtomicStream(targetFile, attrs: FileAttributes)` (unless there are other explanation I did not see).  

We will no longer need `case(targetFile, attrs) => {}`  and usage will be simpler also without additional arrow for pairs `targetFile -> attrs`, but just `withAtomicStream(targetFile, attrs)`.

Reply:
Yes it's mostly preference I guess. I think of this as a single parameter that you always pass at once (i.e. you don't curry it), which is why I choose a tuple. Happy to separate the arguments too, if that's more clear :)